examples/soft-kmeans/README.md

```diff
@@ -0,0 +1,108 @@
+
+<img alt="Soft_z analysis" src="./images/soft-z_samples_sc2.png"/>
+
+The leftward point at `x = -0.687`, shows a distribution of higher soft Z values for the cluster 2, that is clearly distinct for the distribution of soft Z values for cluster 1.
+Therefore, it can be unequivocally assigned to the cluster 2.
+Similarly, the rightward point at `x = 2.369` can be unequivocally assigned to cluster 1.
+However, the middle point at `x = 1.4` displays two distributions of soft Z values that are superimposed.
+This highlights the fact that this point is in an ambiguous area, and cannot be assigned confidently to any of the two cluster.
+Note that this uncertainty is not quantified in the traditional "hard" k-means algorithm, whereas it is available here, through these soft Z samples.
+
+This being said, the soft Z distributions can be summarized to their median value, and consequently, the soft k-means clustering can be converted to a "hard" k-means by assigning each point to its cluster of higher median probability.
+
+<img alt="cluster assignment single chain" src="./images/cluster_assignment_sc_initstate-1-0.png" height="400"/>
+
+In this case, because the lower probability mode of the posterior is sampled, the cluster assignment is rather imperfect.
+The points at the frontier of the 2 clusters (framed in dotted line) were assigned to the left cluster, which is the less probable one.
+
+
+
+## Chainsail sampling
+
+### Sampling
+We also sampled the posterior distribution with Chainsail.
+This time, replica exchange allows the main MCMC chain to not get stuck in one mode, and to sample both the low and the high probability modes.
+
+<img alt="sampling chainsail with initial state = (1, 0)" src="./images/sampling_chainsail_initstate-1-0.png" height="400"/>
+
+The sampled means are shown below.
+The two higher peaks correspond to the samples from the higher posterior mode, and the lower peaks to samples from the lower one.
+Similarly to the samples from the single chain, the means sampled from the lower posterior mode are offset from their expected position, but more importantly, samples from the higher posterior mode match the theoretical cluster centroids much better.
+
+<img alt="sampled means chainsail" src="./images/samples_means_chainsail_initstate-1-0.png" height="400"/>
+
+
+### Cluster assignment
+Similarly, the soft Z samples show bimodal distributions.
+
+<img alt="Soft_z analysis" src="./images/soft-z_samples_chainsail2.png"/>
+
+And finally, the clusters assignments are closer to what would be expected with a hard k-means clustering.
+
+<img alt="cluster assignment chainsail" src="./images/cluster_assignment_chainsail_initstate-1-0.png" height="400"/>
+
+
+
+
+## Reproduce the example
+- The code to generate those figures lives in the jupyter notebook `soft-kmeans.ipynb`. It is packaged with [Nix](https://github.com/NixOS/nix) through [JupyterWith](https://github.com/tweag/jupyterWith). It can be run with the following command:
```

Comment:
```suggestion
- The code to generate those figures lives in the Jupyter notebook `soft-kmeans.ipynb`. It is packaged with [Nix](https://github.com/NixOS/nix) through [JupyterWith](https://github.com/tweag/jupyterWith). It can be run with the following command:
```

Reply:
Let's add an (absolute) link to the example directory in this repository; after all chances are that this post will be published on the Tweag website, which removes the context of the `chainsail-resources` repository (although the post + notebook can still live in there).